client/my-sites/plugins/plans/index.tsx

```diff
@@ -110,6 +132,69 @@ const Plans = ( { intervalType }: { intervalType: 'yearly' | 'monthly' } ) => {
 					flowName={ MARKETPLACE_FLOW }
 					shouldShowPlansFeatureComparison
 					isReskinned
+					customRender={ {
```

Comment:
The click handlers are already exposed, needing custom buttons here is only required because CTAButton is doing too much.

Can I suggest separating the eligibility and warnings modal from the `CTAButton` into a new component that can be used both here and in `CTAButton`? (or at least just here)

Reply:
Yeah, CTAButton is doing too much. I liked the idea of removing eligibility warnings from CTAButton, it would simplify this screen a lot.  Follow-up issue [opened here](https://github.com/Automattic/wp-calypso/issues/68221)

Reply:
Can we do that first? or when will we be doing the refactor?

Reply:
[I did a p2 about this refactor here](https://explorersp2.wordpress.com/2022/09/26/refactoring-ctabutton-component/) @lsl

Reply:
I did a P2 about this refactor here @lsl: pb5gDS-25u-p2